src/test/java/org/springframework/data/r2dbc/core/R2dbcEntityTemplateUnitTests.java

```diff
@@ -460,7 +461,8 @@ void updateShouldInvokeCallback() {
 				Parameter.from("before-save"));
 	}
 
-	@ToString
+	@Value
+	@With
```

Comment:
#### Why did I change this?
This was the main reason for the test not catching the bug. The problem was that `onBeforeConvert(..)` was mutating the instance of the `entity` passed to the callback. This side-effect was spread through the code and reached the private `doUpdate(T entity, SqlIdentifier tableName)` method, making the test think that it was using the value returned by the callback when it was actually using the mutated `entity` passed through the parameter.

I noticed that other parts of this test are already using immutability through `@Value` and `@With`, so I thought it'd make sense to use the same for `Person` with the minimum possible changes to preserve the behavior of the tests 🙂